### PR TITLE
KRACOEUS-8881:Support for MySQL5.6/MariaDB10

### PR DESCRIPTION
**Pull request by blackcathacker**
This primarily involved making sure to drop the foreign key before modifying a referenced column and limited indexes to 767 bytes.
In order to still support Mysql5.5 you also have to make sure that when dropping the foreign keys the names of the contrainsts match exactly.

This once again, rewrites Flyway scripts and will require manual intervention to make sure kc-qa and cx-kc6 don't fail tomorrow. So let me and/or Travis know before merging.

### CODE REVIEW

**Comment by leoherbie** [comment]
It looks like you are fixing mismatched fk lengths here and limiting the size of indexes.  Is that correct?

**Comment by blackcathacker** [comment]
Yep, mostly anyway. Limiting the size of indexes and making sure we don't change the size of a column that is part of a foreign key, which 5.6/10 doesn't allow. I think a lot of those scripts were trying to fix the mismatched key lengths, its just 5.6/10 won't allow changes if a foreign key exists. Also in one location re-creating a foreign key because the tables were renamed and that was causing some issues as well.

File: coeus-db/coeus-db-sql/src/main/resources/co/kuali/coeus/data/migration/sql/mysql/kc/bootstrap/V510_136__KC_IX_KRACOEUS-6157.sql [hunk 1]
```diff
@@ -74,7 +74,7 @@ CREATE INDEX IX_PROP_SCIENCE_KEYWORD_PID ON PROPOSAL_SCIENCE_KEYWORD( PROPOSAL_I
 /
 CREATE INDEX IX_PROP_SPECIAL_REVIEW_PID ON PROPOSAL_SPECIAL_REVIEW( PROPOSAL_ID )
 /
-create index IX_AWD_CUST_DATA_VAL on AWARD_CUSTOM_DATA (VALUE)
+create index IX_AWD_CUST_DATA_VAL on AWARD_CUSTOM_DATA (VALUE(100))
```
Comment (leoherbie) [inline coeus-db/coeus-db-sql/src/main/resources/co/kuali/coeus/data/migration/sql/mysql/kc/bootstrap/V510_136__KC_IX_KRACOEUS-6157.sql]: Why did you choose 100 as the value?

File: coeus-db/coeus-db-sql/src/main/resources/co/kuali/coeus/data/migration/sql/mysql/kc/bootstrap/V510_136__KC_IX_KRACOEUS-6157.sql [hunk 2]
```diff
@@ -34,7 +34,7 @@ create index IX_AWD_CUST_DATA_AWARD_ID on AWARD_CUSTOM_DATA (AWARD_ID)
 /
 create index IX_TNM_DOCUMENT_AWARD_NUMBER ON TIME_AND_MONEY_DOCUMENT( AWARD_NUMBER )
 /
-create index IX_VERSION_HISTORY_CMPST on VERSION_HISTORY (SEQ_OWNER_CLASS_NAME, SEQ_OWNER_VERSION_NAME_VALUE, VERSION_STATUS)
+create index IX_VERSION_HISTORY_CMPST on VERSION_HISTORY (SEQ_OWNER_CLASS_NAME(200), SEQ_OWNER_VERSION_NAME_VALUE, VERSION_STATUS)
```
Comment (leoherbie) [inline coeus-db/coeus-db-sql/src/main/resources/co/kuali/coeus/data/migration/sql/mysql/kc/bootstrap/V510_136__KC_IX_KRACOEUS-6157.sql]: why 200?  I assume you are approaching the max index length?

Comment (blackcathacker) [inline coeus-db/coeus-db-sql/src/main/resources/co/kuali/coeus/data/migration/sql/mysql/kc/bootstrap/V510_136__KC_IX_KRACOEUS-6157.sql]: Approaching. I should have looked it up. The max with a varchar is 255. 200 seems sufficient for a custom data value though. In fact with value, it should probably be shorter as the majority of them are much shorter. But thoughts?

File: coeus-db/coeus-db-sql/src/main/resources/co/kuali/coeus/data/migration/sql/mysql/kc/bootstrap/V510_136__KC_IX_KRACOEUS-6157.sql (hunk 1 above)
Comment (blackcathacker) [inline coeus-db/coeus-db-sql/src/main/resources/co/kuali/coeus/data/migration/sql/mysql/kc/bootstrap/V510_136__KC_IX_KRACOEUS-6157.sql]: Approaching. I should have looked it up. The max with a varchar is 255. 200 seems sufficient for a custom data value though. In fact with value, it should probably be shorter as the majority of them are much shorter. But thoughts?

**Comment by leoherbie** [comment]
```
CREATE INDEX part_of_name ON customer (name(10));
If names in the column usually differ in the first 10 characters, this index should not be much slower than an index created from the entire name column. Also, using column prefixes for indexes can make the index file much smaller, which could save a lot of disk space and might also speed up INSERT operations.
```

I think we are ok.  By my understanding if the value exceeds the index than the index wouldn't be used.  This would only happen as the values grow pretty large.

**Pull request merged**